service/state/core_access.go

```diff
@@ -118,3 +137,22 @@ func (ca *CoreAccessor) SubmitTxWithBroadcastMode(
 	}
 	return txResp.TxResponse, nil
 }
+
+func (ca *CoreAccessor) Transfer(
+	ctx context.Context,
+	to AccAddress,
+	amount Int,
+	gasLim uint64,
+) (*TxResponse, error) {
```

Comment:
You can still use here an Address, but you would need to do:
```suggestion
	addr Address,
	amount Int,
	gasLim uint64,
) (*TxResponse, error) {
to, ok := addr.(types.AccAddress)
if !ok {
	return nil, fmt.Errof("state: unsupported address type")
}
```